pkg/util/deployment/deployment.go

```diff
@@ -131,19 +132,33 @@ func GetReplicaCountForRCs(replicationControllers []*api.ReplicationController)
 }
 
 // Returns the number of available pods corresponding to the given RCs.
-func GetAvailablePodsForRCs(c client.Interface, rcs []*api.ReplicationController) (int, error) {
-	// TODO: Use MinReadySeconds once https://github.com/kubernetes/kubernetes/pull/12894 is merged.
+func GetAvailablePodsForRCs(c client.Interface, rcs []*api.ReplicationController, minReadySeconds int) (int, error) {
 	allPods, err := getPodsForRCs(c, rcs)
 	if err != nil {
 		return 0, err
 	}
+	return getReadyPodsCount(allPods, minReadySeconds), nil
+}
+
+func getReadyPodsCount(pods []api.Pod, minReadySeconds int) int {
 	readyPodCount := 0
-	for _, pod := range allPods {
+	for _, pod := range pods {
 		if api.IsPodReady(&pod) {
-			readyPodCount++
+			// Check if we've passed minReadySeconds since LastTransitionTime
+			// If so, this pod is ready
+			allConditionsReady := true
```

Comment:
This is not the natural way to write it.
allConditionsReady should be false by default.
We should flip it to true if all conditions are satisfied.